extensions/azurecore/src/azureResource/commands.ts

```diff
@@ -181,17 +181,22 @@ export function registerAzureResourceCommands(appContext: AppContext, azureViewT
 		vscode.commands.executeCommand('workbench.actions.modal.linkedAccount');
 	});
 
-	vscode.commands.registerCommand('azure.resource.connectsqlserver', async (node?: TreeNode) => {
+	vscode.commands.registerCommand('azure.resource.connectsqlserver', async (node?: TreeNode | azdata.ObjectExplorerContext) => {
 		if (!node) {
 			return;
 		}
-
-		const treeItem: azdata.TreeItem = await node.getTreeItem();
-		if (!treeItem.payload) {
-			return;
+		let connectionProfile: azdata.IConnectionProfile = undefined;
+		if (node instanceof TreeNode) {
+			const treeItem: azdata.TreeItem = await node.getTreeItem();
+			if (!treeItem.payload) {
+				return;
+			}
+			// Ensure connection is saved to the Connections list, then open connection dialog
+			connectionProfile = Object.assign({}, treeItem.payload, { saveProfile: true });
+		} else if (node.isConnectionNode) {
+			node.connectionProfile = connectionProfile;
```

Comment:
Wait a sec - this seems backwards. Shouldn't it be `connectionProfile = node.connectionProfile`? connectionProfile is always undefined here.

Reply:
😐  🔫 
fixing